### PR TITLE
Handle cluster setup

### LINKED ISSUE #111: When installing a local app without enough permissions the message shown is not clear.
### Steps to reproduce

1.  Download an app from the marketplace in .tar.gz format.
2. Use market in the CLI to install it locally but not having enough permissions on the file.

### Expected Behaviour
A message saying, not enough permissions to read the app file. Or something similar. "Market cannot read app contents" for example.

### Current Behaviour

It tries to access info.xml and fails, saying  "App does not provide an info.xml file".

```
/opt/owncloud# sudo -u www-data ./occ market:install --local=~/guests-0.3.tar.gz
: App does not provide an info.xml file
```

cc @VicDeo

### LINKED ISSUE #117: Add market:uninstall command to occ
This option, available using the web UI, is missing in the command line.

@pmaier1 @hurradieweltgehtunter

### PR DESCRIPTION
**Pull request by DeepDiver1975**
- fixes #111 
- fixes #117 

Blocker - needs 10.0.3 release upfront - see https://github.com/owncloud/core/pull/28543

### CODE REVIEW

**Comment by VicDeo** [comment]
@DeepDiver1975 tests? ;)

**Comment by DeepDiver1975** [comment]
This is ready to be reviewed and merged as soon as we did release 10.0.3

**Comment by PVince81** [comment]
> This is ready to be reviewed and merged as soon as we did release 10.0.3

So does it need to wait or should it be released as part of 10.0.3 ? 😕

**Comment by DeepDiver1975** [comment]
> So does it need to wait or should it be released as part of 10.0.3 ? 😕

should be part of 10.0.3 from my pov - I could relax the dependency and use some defensive coding ... method_exists ....

**Comment by PVince81** [comment]
so we should get this merged then

File: lib/MarketService.php [hunk 1]
```diff
@@ -580,4 +591,13 @@ public function requestLicenseKey() {
 
 		return $demoLicenseKey;
 	}
+
+	public function canInstall() {
+		if (!method_exists($this->appManager, 'canInstall')) {
```
Comment (PVince81) [inline lib/MarketService.php]: add `TODO: remove once min-version is set to 10.0.3+`

**Comment by PVince81** [comment]
@DeepDiver1975 backport to stable10 please

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 backport to stable10 please

there is much more stuff missing afaik .... will take care later ...

**Comment by DeepDiver1975** [comment]
damn ... did we backport anything?

**Comment by DeepDiver1975** [comment]
https://github.com/owncloud/market/pull/131

**Comment by patrickjahns** [comment]
Shouldn't this also address #123  (no update when directory not writeable)

**Comment by DeepDiver1975** [comment]
> Shouldn't this also address #123 (no update when directory not writeable)

yes - closed - thx

**Pull request merged**